core/unit_test/cuda/TestCuda_Spaces.cpp

```diff
@@ -375,4 +375,73 @@ TEST(cuda, impl_view_texture) {
   TestViewCudaTexture<Kokkos::CudaUVMSpace>::run();
 }
 
+// couldn't create a random-access subview of a view of const T in Kokkos::Cuda
+namespace issue_5594 {
+
+template <typename View>
+struct Iota {
+  Iota(const View &view) : view_(view) {}
+  KOKKOS_INLINE_FUNCTION
+  void operator()(int i) const { view_(i) = i; }
+  View view_;
+};
+
+template <typename V1, typename V2>
+struct Checker {
+  Checker(const V1 &v1) : v1_(v1) {}
+  KOKKOS_INLINE_FUNCTION
+  void operator()(int i, bool &lerr) const {
+    V2 v2(&v1_(0), v1_.size());  // create subview in execution space
```

Comment:
This is the line which failed before I assume?
If so: lets just rename this checker to: Issue5594Functor, mark with a comment the line which actually failed before. 
You also need to do a `lerr |= (v1_(i) != v2(i))` or just sum up errors, i.e. do a sum with an integer error counter, which most tests do. (I think we would prefer the error counter).